.github/workflows/tag-publish.yml

```diff
@@ -1,121 +1,80 @@
-name: Tag, Build, and Push Image
+name: Tag, Publish, and Deploy
 # The purpose of the workflow is to:
-#  1. Bump the version number
-#  2. Build and publish the build to Artifactory
+#  1. Bump the version number and tag the release
+#  2. Build and publish the client to Artifactory
 #  3. Build docker image and publish to GCR
 #  4. Trigger deployment to the dev environment
 #
-# When run on merge to dev, it bumps the patch version
+# When run on merge to dev, it tags and bumps the patch version by default. You can
+# bump other parts of the version by putting #major, #minor, or #patch in your commit
+# message.
+#
+# When run on a hotfix branch, it tags and generates the hotfix version
+#
 # When run manually, you can specify the part of the semantic version to bump
+#
+# The workflow relies on github secrets:
+# - REPO_ACCESS_TOKEN - the broadbot token, so we can avoid two reviewer rule on GHA operations
+# - GCR_PUBLISH_KEY   - key and email for publishing the docker to GCR
+# - GCR_PUBLISH_EMAIL
+# - ARTIFACTORY_USERNAME - username and password for publishing the client to artifactory
+# - ARTIFACTORY_PASSWORD
 on:
-#  push:
-#    branches:
-#    - dev
-#    paths-ignore:
-#    - 'README.md'
-#    - '.github/**'
-#    - 'local-dev/**'
+  push:
+    branches:
+    - dev
+    paths-ignore:
+    - 'README.md'
+    - '.github/**'
+    - 'local-dev/**'
   workflow_dispatch:
     inputs:
       bump:
         description: 'Part of the version to bump: major, minor, patch'
         required: false
         default: 'patch'
+      branch:
+        description: 'Branch to run the workflow on'
+        required: false
+        default: 'main'
```

Comment:
There isn't a branch with this name. Should it be dev?

Reply:
yup